roles/aap_ocp_install/tasks/pre-validate-platform.yml

```
@@ -0,0 +1,198 @@
+  block:
+    - name: Ensure controller garbage collect secrets is valid
+      ansible.builtin.assert:
+        that:
+          - (aap_ocp_install_controller['garbage_collect_secrets'] | string | lower) is in ['true', 'false']
+        quiet: true
+  rescue:
+    - name: Update validation errors fact - controller garbage_collect_secrets
+      ansible.builtin.set_fact:
+        __aap_ocp_install_prevalidate_errors: >
+          {{ __aap_ocp_install_prevalidate_errors + ["aap_ocp_install_controller['garbage_collect_secrets'] must be one of: true or false"] }}
+
+- name: Ensure controller projects persistence is valid (block)
+  when: aap_ocp_install_controller['projects_persistence'] is defined
+  block:
+    - name: Ensure controller projects persistence is valid
+      ansible.builtin.assert:
+        that:
+          - (aap_ocp_install_controller['projects_persistence'] | string | lower) is in ['true', 'false']
+        quiet: true
+  rescue:
+    - name: Update validation errors fact - controller projects_persistence
+      ansible.builtin.set_fact:
+        __aap_ocp_install_prevalidate_errors: >
+          {{ __aap_ocp_install_prevalidate_errors + ["aap_ocp_install_controller['projects_persistence'] must be one of: true or false"] }}
+
+- name: Ensure controller replicas is valid (block)
+  when: aap_ocp_install_controller['replicas'] is defined
+  block:
+    - name: Ensure controller replicas is valid
+      ansible.builtin.assert:
+        that:
+          - (aap_ocp_install_controller['replicas'] | int) > 0
+        quiet: true
+  rescue:
+    - name: Update validation errors fact - controller replicas
+      ansible.builtin.set_fact:
+        __aap_ocp_install_prevalidate_errors: >
+          {{ __aap_ocp_install_prevalidate_errors + ["aap_ocp_install_controller['replicas'] must be a number greater than 0"] }}
+
+- name: Ensure controller projects storage size is valid (block)
+  when: aap_ocp_install_controller['projects_storage_size'] is defined
+  block:
+    - name: Ensure controller projects storage size is valid
+      ansible.builtin.assert:
+        that:
+          - (aap_ocp_install_controller['projects_storage_size'] | regex_replace('^(?P<size>[0-9]+)(?P<si>Ki|K|Mi|M|Gi|G|Ti|T|Pi|P|Ei|E)$', '\\g<size>') | int) > 0
+          - (aap_ocp_install_controller['projects_storage_size'] | regex_replace('^(?P<size>[0-9]+)(?P<si>Ki|K|Mi|M|Gi|G|Ti|T|Pi|P|Ei|E)$', '\\g<si>')) is in ['Ki','K','Mi','M','Gi','G','Ti','T','Pi','P','Ei','E']
+        quiet: true
+  rescue:
+    - name: Update validation errors fact - projects_storage_size replicas
+      ansible.builtin.set_fact:
+        __aap_ocp_install_prevalidate_errors: >
+          {{ __aap_ocp_install_prevalidate_errors + ["aap_ocp_install_controller['projects_storage_size'] must be a number greater than 0 with a size (e.g. 12Gi or 10000M)"] }}
+
+- name: Ensure hub storage type is valid (block)
+  when: aap_ocp_install_hub is defined
```

If the intention is to default the `storage_type` to `file` (which I think is a good default). Then this when clause should only be checked when a `storage_type` is defined.

```suggestion
  when: aap_ocp_install_hub['storage_type'] is defined
```